### ISSUE TITLE
nested form not saved to db

### ISSUE THREAD

**Issue by worldbest212** [nathanvda/cocoon#378]
Hi. 

I searched my error in google and here, but still I can't not find solution.

when I use cocoon in different project, there is no problem(haml)
but in this time with erb I got the error as mentioned title

View/new.html.erb

        <%= simple_form_for @project, :html => { class: 'dd-form floating-labels'}  do |f| %>
	<%= f.input :project_id, label: 'Project ID', :label_html=> { class: 'cd-label' },input_html: { for: 'cd-name', class: 'user'}   %>
  							
  	<%= f.input :name, label: 'Project Name', label_html: { class: 'cd-label' },input_html: { for: 'cd-name', class: 'company'} %>
  	<%= f.collection_radio_buttons :shaft,[['Single', 'Single'] ,['Multi(2+1)', 'Multi(2+1)'],['Multi(1+1)', 'Multi(1+1)']],:first, :last %>

	<%= f.input_field :number, :collection => { '1' => 1, '2' => 2,'3' => 3,'4' => 4,'5' => 5,'6' => 6,'7' => 7,'8' => 8}, class:'number', label: false  %>

      <%= f.input :text, label: 'Project description', label_html: { class: 'cd-label' },input_html: { for: 'cd-textarea', class: 'message'}   %>
	<%= f.input_field :currency, :collection => { 'Dollar' => 1, 'Euro' => 2,'KRW' => 3}, class:'currency', label: false  %>
	<%= f.input_field :area, :collection => { 'Asia' => 1, 'Middle East' => 2,'America' => 3,'Europe' => 4,'Africa' => 5}, class:'area', label: false  %>
        <%= f.input :location,  label: 'Location', label_html: { class: 'cd-label' },input_html: { for: 'cd-name', class: 'location'} %>

  	<h2>Equipment Cost Caculation</h2>
						<p>

							<table class="table table-condensed table-hover table-responsive">
							<thead>
							<tr><th></th>	<th>Equipment</th>	<th>Type</th> <th>Currency</th>	<th>Qty</th>		<th>Unit price</th>	
							</tr>
							</thead>
					
							<tbody id="equips">
							
  							<%= f.simple_fields_for  :equips do |equip| %>
							<%= render  'equip_fields',  f: equip  %>
							<% end %>
							</tbody>
							
							
							</table>
							<div class='links'> <%= link_to_add_association 'add equipment', f, :equips, data: {"association-insertion-node" => "tbody#equips", "association-insertion-method" => "append"} %></div>
							
						</p>

view/projects/_equip_fields.html.erb

  	
        <tr class="nested-fields">
	<td><%= link_to_remove_association "X", f %></td>
	<td><%= f.input :category_id, collection: Category.all, prompt: "Choose a Equipment", label: false %></td>
        <td><%= f.input(:subcategory_id, collection: Category.where(:id => [1,2,3]), as: :grouped_select, group_method: :subcategories, label: false, group_label_method: :name,:label_method =>:name, prompt: "Choose a Type") %></td>
	<td><%= f.input :currency, :collection => { 'Dollar' => 1, 'Euro' => 2,'KRW' => 3}, class:'shorttd', label: false  %></td>
	<td><%= f.input :quantity, label: false, :class => 'shorttd' %></td>

	<td><%= f.input :unit_price, label: false %></td>
</tr>

project.rb

         class Project < ApplicationRecord

	   private

	    has_many :equips
	    accepts_nested_attributes_for :equips,reject_if: :all_blank, allow_destroy: true

	     belongs_to :user

                 end

equip.rb

             class Equip < ApplicationRecord
	               belongs_to :category	
                       belongs_to :subcategory
                      belongs_to :project

            end

project.contoller

	def new
		@project = current_user.projects.build
		render  :layout => false
	end

	def create

		@project = current_user.projects.build(project_params)
		if @project.save
				redirect_to @project, notice: 'project was successfully created.'
		else
			render 'new' 
		end
	end

	private

	  def project_params
		params.require(:project).permit(:project_id, :name, :area, :location, :number, :text, :shaft, :total_price,:currency, 
			equips_attributes: [:id, :_destroy, :category_id, :subcategory_id, :quantity, :currency, :unit_price, :price, :project_id])
	 end

**Comment by nathanvda** [comment]
* please format your html/code to be legible (=use consistent indentation)
* Please clarify the error: if you get an error-message, can you show it? Or you can edit without any apparent error, but the nested data is not saved or the entire form is not saved? You can't create new items or you can't edit? 
* please show what is posted to the controller (it is in your logfile)

**Comment by nathanvda** [comment]
Please show what is posted to the controller. This will help to clarify if your strong parameters definition is correct.

Also really please fix up the indentiation on all code (in this question, frankly I don't care how you write your own code, but for me it is really hard to follow any flow and look for errors). 

Why is there a `private` in the `Project` class?

**Comment by worldbest212** [comment]
The below is log file.

Started GET "/projects/new" for 128.134.5.47 at 2016-08-07 14:07:52 +0000
Cannot render console from 128.134.5.47! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
Processing by ProjectsController#new as HTML
  [1m[36mUser Load (0.4ms)[0m  [1m[34mSELECT  "users".* FROM "users" WHERE "users"."id" = ? ORDER BY "users"."id" ASC LIMIT ?[0m  [["id", 3], ["LIMIT", 1]]
  Rendering projects/new.html.erb
  [1m[36mCategory Load (0.1ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  [1m[36mCategory Load (0.1ms)[0m  [1m[34mSELECT "categories".* FROM "categories" WHERE "categories"."id" IN (1, 2, 3)[0m
  [1m[36mSubcategory Load (0.1ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 1]]
  [1m[36mSubcategory Load (0.2ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 2]]
  [1m[36mSubcategory Load (0.1ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 3]]
  Rendered projects/_equip_fields.html.erb (12.2ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_pm_fields.html.erb (5.4ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_en_fields.html.erb (5.3ms)
  Rendered projects/new.html.erb (72.9ms)
Completed 200 OK in 77ms (Views: 73.1ms | ActiveRecord: 1.1ms)


Started POST "/projects" for 128.134.5.47 at 2016-08-07 14:08:38 +0000
Cannot render console from 128.134.5.47! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
Processing by ProjectsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"n46dPcOPc6P+dDocBq2DH/slbyH0NWjaZMUTJqvyjC6UneIw/Gsfh3qR7gmUKhJq77+akyGWols6UJPeOJeClA==", "project"=>{"project_id"=>"kr1214", "name"=>"4414", "shaft"=>"Single", "number"=>"3", "text"=>"1414", "currency"=>"2", "area"=>"2", "location"=>"1414"}, "commit"=>"Save"}
  [1m[36mUser Load (0.2ms)[0m  [1m[34mSELECT  "users".* FROM "users" WHERE "users"."id" = ? ORDER BY "users"."id" ASC LIMIT ?[0m  [["id", 3], ["LIMIT", 1]]
  [1m[35m (0.1ms)[0m  [1m[36mbegin transaction[0m
  [1m[35mSQL (0.6ms)[0m  [1m[32mINSERT INTO "projects" ("project_id", "name", "area", "shaft", "currency", "location", "created_at", "updated_at", "text", "number", "user_id") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)[0m  [["project_id", "kr1214"], ["name", "4414"], ["area", "2"], ["shaft", "Single"], ["currency", "2"], ["location", "1414"], ["created_at", 2016-08-07 14:08:38 UTC], ["updated_at", 2016-08-07 14:08:38 UTC], ["text", "1414"], ["number", 3], ["user_id", 3]]
  [1m[35m (5.9ms)[0m  [1m[36mcommit transaction[0m
Redirected to http://52.42.5.108:3000/projects/37
Completed 302 Found in 13ms (ActiveRecord: 6.8ms)

**Comment by nathanvda** [comment]
From the log in the `POST` part you can see what is posted: 

    "project"=>{"project_id"=>"kr1214", "name"=>"4414", "shaft"=>"Single", "number"=>"3", "text"=>"1414", "currency"=>"2", "area"=>"2", "location"=>"1414"}, "commit"=>"Save"}

No nested data there, so that is explains why it is not saved. Probably something wrong with your HTML, e.g. your nested data is _outside_ the form?

**Comment by worldbest212** [comment]
Thank you for your response. I finally realize what I mistake in HTML code. form is under <li> tag and nested tag is under another <li> tag. That means <% end %> end except nested fields. 

I cleaned up tags to check situation. And then I can see the parameters but still post is not saved and just render 'new' according to controller. 

Processing by ProjectsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"Huwj+/uWEmbQPboHjaq9yNDUCmGVj1NYOm9qeUzuE80V/1z2xHJ+QlTYbhIfLSy9xE7/00Asmdlk+uqB34sddw==", "project"=>{"project_id"=>"141414", "name"=>"141414", "shaft"=>"Single", "number"=>"5", "text"=>"141414515", "currency"=>"1", "area"=>"1", "location"=>"151515", "equips_attributes"=>{"1470582086454"=>{"_destroy"=>"false", "category_id"=>"1", "subcategory_id"=>"3", "currency"=>"2", "quantity"=>"1", "unit_price"=>"1"}}, "pms_attributes"=>{"1470582102026"=>{"_destroy"=>"false", "category_id"=>"1", "currency"=>"1", "mh"=>"1", "unit_price"=>"1"}}, "ens_attributes"=>{"1470582112052"=>{"_destroy"=>"false", "category_id"=>"1", "currency"=>"2", "mh"=>"1", "unit_price"=>"1"}}}, "commit"=>"Save"}
  [1m[36mUser Load (0.4ms)[0m  [1m[34mSELECT  "users".* FROM "users" WHERE "users"."id" = ? ORDER BY "users"."id" ASC LIMIT ?[0m  [["id", 3], ["LIMIT", 1]]
  [1m[35m (0.1ms)[0m  [1m[36mbegin transaction[0m
  [1m[36mCategory Load (0.1ms)[0m  [1m[34mSELECT  "categories".* FROM "categories" WHERE "categories"."id" = ? LIMIT ?[0m  [["id", 1], ["LIMIT", 1]]
  [1m[36mSubcategory Load (0.1ms)[0m  [1m[34mSELECT  "subcategories".* FROM "subcategories" WHERE "subcategories"."id" = ? LIMIT ?[0m  [["id", 3], ["LIMIT", 1]]
  [1m[36mCategory Load (0.0ms)[0m  [1m[34mSELECT  "categories".* FROM "categories" WHERE "categories"."id" = ? LIMIT ?[0m  [["id", 1], ["LIMIT", 1]]
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT  "categories".* FROM "categories" WHERE "categories"."id" = ? LIMIT ?[0m  [["id", 1], ["LIMIT", 1]]
  [1m[35m (0.1ms)[0m  [1m[31mrollback transaction[0m
  Rendering projects/new.html.erb within layouts/application
  [1m[36mCategory Load (0.1ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  [1m[36mCategory Load (0.1ms)[0m  [1m[34mSELECT "categories".* FROM "categories" WHERE "categories"."id" IN (1, 2, 3)[0m
  [1m[36mSubcategory Load (0.1ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 1]]
  [1m[36mSubcategory Load (0.2ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 2]]
  [1m[36mSubcategory Load (0.2ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 3]]
  Rendered projects/_equip_fields.html.erb (13.2ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories" WHERE "categories"."id" IN (1, 2, 3)[0m
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 1]]
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 2]]
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "subcategories".* FROM "subcategories" WHERE "subcategories"."category_id" = ?[0m  [["category_id", 3]]
  Rendered projects/_equip_fields.html.erb (10.9ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_pm_fields.html.erb (5.2ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_pm_fields.html.erb (5.1ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_en_fields.html.erb (5.3ms)
  [1m[36mCACHE (0.0ms)[0m  [1m[34mSELECT "categories".* FROM "categories"[0m
  Rendered projects/_en_fields.html.erb (5.3ms)
  Rendered projects/new.html.erb within layouts/application (85.0ms)
Completed 200 OK in 140ms (Views: 99.8ms | ActiveRecord: 1.7ms)

**Comment by worldbest212** [comment]
Thank you for your respond. Without nested field, the submit is not functioned. That means there is another mistake. thank you again!!!
